src/app/Slices/authSlice.js

```diff
@@ -0,0 +1,93 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { toast } from "react-toastify";
+import axios from "axios";
+
+const initialState = {
+  isSignedUp: false,
+  loginError:false,
+  isLoggedIn: localStorage.getItem("token") ? true : false,
+  token: localStorage.getItem("token") || null,
+  user: JSON.parse(localStorage.getItem("user")) || null,
+};
+
+export const loginUser = createAsyncThunk(
+  "auth/loginUser",
+  async ({ email, password }, thunkAPI) => {
+    try {
+      const response = await axios.post("/api/auth/login", {
+        email: email,
+        password: password,
+      });
+      return response.data;
+    } catch (e) {
+      return thunkAPI.rejectWithValue(e);
+    }
+  }
+);
+
+export const signupUser = createAsyncThunk(
+  "auth/signupUser",
+  async ({ data }, thunkAPI) => {
+    try {
+      const response = await axios.post("api/auth/signup", {
+        ...data,
+      });
+      return response.data;
+    } catch (e) {
+      return thunkAPI.rejectWithValue(e);
+    }
+  }
+);
+
+export const authSlice = createSlice({
+  name: "auth",
+  initialState,
+  reducers: {
+    logoutUser: () => {
+      localStorage.removeItem("token");
+      localStorage.removeItem("user");
+      localStorage.removeItem("login");
+      localStorage.removeItem("watchlater");
+      localStorage.removeItem("liked");
+      localStorage.removeItem("history");
+      localStorage.removeItem("playlist");
```

Comment:
Suggestion:- If you are clearing the entire local storage then you can use localStorage.clear() instead of explicitly removing individual items.

Reply:
ok, thank for the suggestion